### ISSUE TITLE
Chrome app: Promise will not resolve and the app hangs

### ISSUE THREAD

**Issue by jarrodek** [pouchdb/pouchdb#5875]
### Issue
Some of users of my app reported that one of the modules of the app isn't working properly. Other users complained about different part of the app not related on a first sight. I checked what is going on and discovered that the PouchDB does not resolve a promise when requesting data from the datastore. It doesn't reject it either.

This is what I can see in the console when I try to run test code:

```
var p = new PouchDB('environments');
pouchdb:adapter Picked adapter: idb +0ms
undefined
p.allDocs().then((a) => console.log(a));
Promise {[[PromiseStatus]]: "pending", [[PromiseValue]]: undefined}
p.allDocs({}).then((a) => console.log(a));
Promise {[[PromiseStatus]]: "pending", [[PromiseValue]]: undefined}
p.allDocs({}).then((a) => console.log(a)).catch((e) => console.error(e));
Promise {[[PromiseStatus]]: "pending", [[PromiseValue]]: undefined}
```
The promise hasn't resolved while I was writing this post.

There's another issue that I think is related. When I initialize the app one of the data stores is not initialized and PouchDb throws an error in function `addDeletedOrLocalIndex`:
```
var docStore = txn.objectStore(DOC_STORE);
// Uncaught DOMException: Failed to execute 'objectStore' on 'IDBTransaction': The specified object store was not found.
```
I first I though that some users temper with the database or something but it happened to me on a test app I never used before so it was clean install.

You can find related issue here: https://github.com/jarrodek/ChromeRestClient/issues/887

### Info
- Environment: Chrome App
- Platform: Chrome
- Adapter: IndexedDB (only available for Chrome Apps)

### Reproduce
I have no idea how to reproduce it. It seems really random. I have dev channel of  the app installed alongside the stable channel on the same browser profile. Right now both are identical. One is working and another is not. I'd like to provide you with more data but I can't figure that one out.
I can only track the code to `toPromise` function and to `self.taskqueue.addTask` line. After that nothing is happening (all functions are returning). It looks like the task queue is not executed.
Moreover it is working when the first time I use the data store. Subsequent calls tho the same data store ends with hanging the app.

**Comment by jarrodek** [comment]
One more thing. When I've changed the line:
```
if (e.oldVersion <= 1) {
  return createSchema(db); // new db, initial schema
}
```
To less or equal 1 it started working. Is this a workaround for this issue?

**Comment by nolanlawson** [comment]
Thanks for raising this issue, however we would need a test to reproduce to understand how to resolve it.

**Comment by jarrodek** [comment]
I'm not sure either. It was very rare and I was lucky that some day it also happened to me. 
Hope you will find the bug.

**Comment by nolanlawson** [comment]
It may be an issue in Chrome IDB itself. I'll leave this open in case anyone can find a repro

**Comment by GaragusoPD** [comment]
I have the same issue. PouchDB seemes unable to create the db in Chrome 55. I get these errors:
```
pouchdb-6.1.1.min.js:8 Uncaught DOMException: Failed to execute 'index' on 'IDBObjectStore': The transaction has finished.
    at IDBRequest.u (http://127.0.0.1:59767/pouchdb-6.1.1.min.js:8:18161)
u @ pouchdb-6.1.1.min.js:8
pouchdb-6.1.1.min.js:7 Database has a global failure DOMException: Transaction timed out due to inactivity.
```

To reproduce it, I use this simple file:

```
<!document html>
<html>
	<head>
	</head>
	<body>
		<h1>Test ground.</h1>
		<p>Open the console to test javascript</p>
		<p id="count"></p>
		<script src="pouchdb-6.1.1.min.js"></script>
		<script>
			var db=new PouchDB('testDB',{auto_compaction: true});
			var p=document.getElementById('count');
			
			// Create records
			var start, end, docs;
			
			
			Start();			
			console.info('Adding 100 000 records');
			for(i=0; i<100000; i++){
			
				db
				.post({title: 'document:'+i})
				.then(function(data){
					p.innerHTML=i;
				});
			
			}
			End();
			
			
			
			console.info('Retrieving documents');
			Start();
			db
				.allDocs({include_docs:true})
				.then(function(result){
					End();
					console.log('Total rows: ' + result.total_rows);
					console.log(result.rows);
				});		
			
					
			function Start(){
				start=new Date();
				//console.log(start);
			}
			
			function End(){
				console.info('Process complete.');
				end=new Date();
				//console.log(end);
				console.log('Miliseconds: ' + (end-start));
			}
		</script>
	</body>
</html>
```

**Comment by nolanlawson** [comment]
This sounds like it may be a bug in Chrome itself, especially if you cannot reproduce in any other browsers. A test to reproduce would be extremely helpful, or at the very least a full stacktrace.

@GaragusoPD it seems almost certain that you have a separate issue because your error message (`Transaction timed out due to inactivity`) is different from @jarrodek 's error message (`The specified object store was not found.`). My hunch is that the database got corrupted somehow; have you tried testing on a different machine or in a different version of Chrome (e.g. Chrome Canary) or wiping data and trying again?

**Comment by GaragusoPD** [comment]
I did test it in Chrome and Firefox, with similar results. I kind of "solved" it and have it working now. I'm just guessing now as to why it happens with that particular code. It seems to me that Chrome needs some time to initialize IndexedDB and/or the long chain of requests (maybe some kind of optimization process?). It seems to cache all the promises before starting working on them, and if the number is too high, it crashes. The .bulkDocs() doesn't have that issue. Either way, I suggest avoiding posting documents inside a loop with large number of documents or use .bulkDocs() instead with the same limit. I was able to test up to 600 000 documents in Chrome, and found that to be an absolute crashing number. If you are interested, It has worked well with batches of up to 100 000 documents (even though it takes time and kind of freezes the UI for some time).

Thanks @nolanlawson for your quick reply!

**Comment by jpike88** [comment]
@GaragusoPD How did you 'solve' this issue?

**Comment by jpike88** [comment]
Just encountered this issue on a customer's computer. Windows 10, Chrome 58.0.3029.110

Absolute pain in the butt to debug, same issue as OP.

**Comment by jarrodek** [comment]
@jpike88 See my comment here: https://github.com/pouchdb/pouchdb/issues/5875#issuecomment-260361336
This helped me.

**Comment by jpike88** [comment]
Interesting fact: I got the customer to install Chrome canary and the issue persisted. I'm almost inclined to blame Windows 10 or some registry related crap.

**Comment by jpike88** [comment]
The customer also had malware so could have been that too
